arborist/server_test.go

```diff
@@ -2341,15 +2493,204 @@ func TestServer(t *testing.T) {
+					assert.Contains(t, result, resource, msg)
+					assert.ElementsMatch(t, result[resource], actions, msg)
+				}
+			})
+
+			t.Run("GETwithJWT_userDoesNotExist", func(t *testing.T) {
+				w := httptest.NewRecorder()
+				url := "/auth/mapping"
+				req := newRequest("GET", url, nil)
+				badUsername := "hulkhogan12"
+				token := TestJWT{username: badUsername}
+				req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", token.Encode()))
+				handler.ServeHTTP(w, req)
+
+				// expect a 200 OK response
+				assert.Equal(t, w.Code, http.StatusOK, "expected a 200 OK")
+
+				// expect result to only contain anonymous and loggedIn auth mappings.
+				result := make(arborist.AuthMapping)
+				err = json.Unmarshal(w.Body.Bytes(), &result)
+				if err != nil {
+					httpError(t, w, "couldn't read response from auth mapping")
+				}
+				expectedMappings := make(arborist.AuthMapping)
+				for k, v := range anonymousAuthMapping {
+					expectedMappings[k] = v
+				}
+				for k, v := range loggedInAuthMapping {
+					expectedMappings[k] = v
+				}
+				msg := fmt.Sprintf("Expected response to be these auth mappings from anonymous and logged-in groups: %v", expectedMappings)
+				for resource, actions := range result {
+					assert.Contains(t, expectedMappings, resource, msg)
+					assert.ElementsMatch(t, expectedMappings[resource], actions, msg)
+				}
+			})
+
+			t.Run("GET_noUsernameProvided", func(t *testing.T) {
+				w := httptest.NewRecorder()
+				url := "/auth/mapping"
+				req := newRequest("GET", url, nil)
+				handler.ServeHTTP(w, req)
+				if w.Code != http.StatusOK {
+					httpError(t, w, "expected to get policies for Anonymous group; got bad response instead")
+				}
+
+				// expect a 200 OK response
+				assert.Equal(t, w.Code, http.StatusOK, "expected a 200 OK")
+
+				// expect result to contain only authMappings of anonymous policies
+				result := make(arborist.AuthMapping)
+				err = json.Unmarshal(w.Body.Bytes(), &result)
+				if err != nil {
+					httpError(t, w, "couldn't read response from auth mapping")
+				}
+				msg := fmt.Sprintf("Expected these auth mappings from anonymous group: %v \t Got: %v", anonymousAuthMapping, result)
+				for resource, actions := range result {
+					assert.Contains(t, anonymousAuthMapping, resource, msg)
+					assert.ElementsMatch(t, anonymousAuthMapping[resource], actions, msg)
+				}
 			})
 
 			t.Run("POST", func(t *testing.T) {
 				w := httptest.NewRecorder()
 				body := []byte(fmt.Sprintf(`{"username": "%s"}`, username))
 				req := newRequest("POST", "/auth/mapping", bytes.NewBuffer(body))
 				handler.ServeHTTP(w, req)
+				// expect to receive user's auth mappings
 				testAuthMappingResponse(t, w)
+				// expect to also receive auth mappings of anonymous and logged-in policies
+				result := make(arborist.AuthMapping)
+				err = json.Unmarshal(w.Body.Bytes(), &result)
+				if err != nil {
+					httpError(t, w, "couldn't read response from auth mapping")
```

Comment:
since this is repeated so many times maybe it would be worth it to update the `testAuthMappingResponse` function to take 2 optional boolean parameters `checkAnonymous` and `checkLoggedIn` or something